core/include/drivers/stm32_tamp.h

```diff
@@ -34,10 +36,37 @@
  * Protection zone 3
  * Backup registers from TAMP_BKP<t>R to last backup register are in zone 3,
  * with <t> = (@nb_zone1_regs1 + @nb_zone2_regs).
+ *
+ * When RIF is supported, each zone can be subdivided to restrain accesses to
+ * some CIDs
  */
 struct stm32_bkpregs_conf {
 	uint32_t nb_zone1_regs;
 	uint32_t nb_zone2_regs;
+	uint32_t *rif_offsets;
+};
+
+/**
+ * struct stm32_tamp_platdata - TAMP platform data
+ * @base: IOMEM base address
+ * @bkpregs_conf: TAMP backup register configuration reference
+ * @compat: Reference to compat data passed at driver initialization
+ * @is_tdcid: Indicates whether this CPU runs as the TDCID
+ * @conf_data: RIF configuration data
+ * @clock: TAMP clock
+ * @nb_rif_resources: Number of RIF resources
+ * @it: TAMP interrupt number
+ * @is_tdcid: True if current processor is TDCID
+ */
+struct stm32_tamp_platdata {
+	struct io_pa_va base;
+	struct stm32_bkpregs_conf *bkpregs_conf;
+	struct stm32_tamp_compat *compat;
```

Comment:
`struct stm32_tamp_compat` definition is missing (and likely preferred only in the scope of *stm32_tamp.c*)

Reply:
Fixed when reorganising the header file